src/Symfony/Component/Console/Application.php

```diff
@@ -276,6 +277,13 @@ public function doRun(InputInterface $input, OutputInterface $output)
         return $exitCode;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function reset()
```

Comment:
what's the point of an empty reset method? why not implement reset only in the frameworkbundle application?

Reply:
Command::getApplication() returns this very class according to the phpdoc. 
With this, phpstan is happy

Reply:
Please, please, please, stop with the phpstan mania. We don't care if phpstan is happy or not. This is just a tool.

Reply:
Actually I did it to make @nicolas-grekas happy (see previouses comment) he is not a tools 😅😁

Reply:
The reason @Tobion is that this method is now part of the semantics of the API exposed by the base `Application` class. It's a template method (from the template design pattern).

Reply:
template pattern does not apply here. for a template pattern methods must either be abstract (to force clients to implement them) or function as ahelper method hooks (with can have an empty body). neither is the case here. it's just an empty method providing no functionality. but it's not that important to me.